codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAccountIdEndpointModeRuntimeConfig.java

```diff
@@ -0,0 +1,122 @@
+
+    @Override
+    public void addConfigInterfaceFields(
+            TypeScriptSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            TypeScriptWriter writer
+    ) {
+        if (isAwsService(settings, model)) {
+            ServiceShape service = settings.getService(model);
+            EndpointRuleSetTrait endpointRuleSetTrait = service.getTrait(EndpointRuleSetTrait.class)
+                    .orElseThrow(() -> new RuntimeException("service missing EndpointRuleSetTrait"));
+            RuleSetParameterFinder ruleSetParameterFinder = new RuleSetParameterFinder(service);
+            if (ruleSetParameterFinder.getBuiltInParams().containsKey("AccountIdEndpointMode")) {
+                writer.writeDocs("Defines if the AWS AccountId will be used for endpoint routing.")
+                        .write("accountIdEndpointMode?: AccountIdEndpointMode | __Provider<AccountIdEndpointMode>;\n");
+            }
+        }
+    }
+
+    @Override
+    public void prepareCustomizations(
+        TypeScriptWriter writer,
+        LanguageTarget target,
+        TypeScriptSettings settings,
+        Model model
+    ) {
+        if (isAwsService(settings, model) && target.equals(LanguageTarget.NODE)) {
+            writer.addDependency(AwsDependency.AWS_SDK_CORE);
+            writer.addImport("emitWarningIfUnsupportedVersion", "awsCheckVersion", AwsDependency.AWS_SDK_CORE);
+            writer.write("awsCheckVersion(process.version);");
+        }
+    }
+
+    @Override
+    public Map<String, Consumer<TypeScriptWriter>> getRuntimeConfigWriters(
+            TypeScriptSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            LanguageTarget target
+    ) {
+        ServiceShape service = settings.getService(model);
+        Map<String, Consumer<TypeScriptWriter>> runtimeConfigs = new HashMap<>();
+        if (isAwsService(settings, model)) {
+            EndpointRuleSetTrait endpointRuleSetTrait = service.getTrait(EndpointRuleSetTrait.class)
+                    .orElseThrow(() -> new RuntimeException("service missing EndpointRuleSetTrait"));
+            RuleSetParameterFinder ruleSetParameterFinder = new RuleSetParameterFinder(service);
+            if (ruleSetParameterFinder.getBuiltInParams().containsKey("AccountIdEndpointMode")) {
+                switch (target) {
+                    case BROWSER:
+                        runtimeConfigs.put("accountIdEndpointMode", writer -> {
+                            writer.addDependency(TypeScriptDependency.CONFIG_RESOLVER);
+                            writer.addImport("DEFAULT_ACCOUNT_ID_ENDPOINT_MODE", "DEFAULT_ACCOUNT_ID_ENDPOINT_MODE",
+                                    TypeScriptDependency.CONFIG_RESOLVER);
+                            writer.write("(() => Promise.resolve(DEFAULT_ACCOUNT_ID_ENDPOINT_MODE))");
```

Comment:
`async () => val`

Reply:
keeping that as is for uniformity with [other similar options](https://github.com/aws/aws-sdk-js-v3/blob/e90775f807f0d4643b976a47f3c401f44f4940f1/codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddAwsRuntimeConfig.java#L202)